src/scf/newton.jl

```diff
@@ -0,0 +1,317 @@
+# Newton's algorithm to solve SCF equations
+#
+# Newton algorithm consist of iterating over density matrices like
+#       P <- P - δP, with additional renormalization
+# where δP solves ([1])
+#       (Ω+K)δP = [P, [P, H(P)]]
+# where (Ω+K) if the constrained Hessian of the energy. It is
+# defined as a super operator on the space tangent to the constraints
+# manifold at point P_∞, the solution of our problem.
+#   - K is the unconstrained Hessian of the energy on the manifold;
+#   - Ω represents the influence of the curvature of the manifold :
+#         ΩδP = -[P_∞, [H(P_∞), δP]].
+#     In practice, we dont have access to P_∞ so we just use the current P.
+#     Another way to see Ω is as the four-point independent-particle
+#     susceptibility. Indeed, if we define the following extension and
+#     retraction operators
+#       - E : builds a density matrix Eρ given a density ρ via Eρ(r,r') = δ(r,r')ρ(r),
+#       - R : builds a density ρ given a density matrix O via ρ(r) = O(r,r),
+#     then we have the relation χ0₂ = -R(Ω₄^-1)E, where χ0₂ is the two_point
+#     independent-particle susceptibility (it returns δρ from a given δV) and
+#     Ω₄ is the four-point independent-particle susceptibility.
```

Comment:
This I find a bit strange. The way I see it \Omega is the negative inverse of the 4-point independent particle susceptibility. Wouldn't you agree?

Reply:
Depends on how you see it. I see Omega as the central object, and chi0 as a derived quantity. In a general context where you solve F(x, lambda) = 0, Omega is dF/dx, and the response operator is -dF/dx^-1 dF/dlambda, so the "primitive" thing is dF/dx.

Reply:
Ok but then nevertheless calling  Ω₄ is the four-point independent-particle susceptibility is confusing, because the relationship is inverse.

Reply:
Oh sorry I didn't read the comment. Of course you're right. chi04 = -Omega4^2 is the 4pt indep-particle susceptibility

Reply:
Not sure that I follow here

Reply:
Just what I said: chi04 = -Omega4^2, and chi02 = R chi04 E. In your comment you wrote Omega4 is the 4p indep-particle susceptibility, but it's chi04

Reply:
Yes, but I don't get what is Omega4^2

Reply:
sorry of course, I meant Omega4^-1